src/function/gds/k_core_decomposition.cpp

```diff
@@ -0,0 +1,378 @@
+        return std::vector<LogicalTypeID>{LogicalTypeID::ANY};
+    }
+    /*
+     * Outputs are
+     *
+     * _node._id::INTERNAL_ID
+     * group_id::INT64
+     */
+    binder::expression_vector getResultColumns(binder::Binder* binder) const override {
+        expression_vector columns;
+        auto& outputNode = bindData->getNodeOutput()->constCast<NodeExpression>();
+        columns.push_back(outputNode.getInternalID());
+        columns.push_back(binder->createVariable(GROUP_ID_COLUMN_NAME, LogicalType::INT64()));
+        return columns;
+    }
+
+    void bind(const GDSBindInput& input, main::ClientContext& context) override {
+        auto graphName = binder::ExpressionUtil::getLiteralValue<std::string>(*input.getParam(0));
+        auto graphEntry = bindGraphEntry(context, graphName);
+        auto nodeOutput = bindNodeOutput(input.binder, graphEntry.nodeEntries);
+        bindData = std::make_unique<GDSBindData>(std::move(graphEntry), nodeOutput);
+    }
+
+    void exec(processor::ExecutionContext* context) override {
+        auto clientContext = context->clientContext;
+        auto mm = clientContext->getMemoryManager();
+        auto graph = sharedState->graph.get();
+        auto numNodesMap = graph->getNumNodesMap(clientContext->getTransaction());
+        auto numThreads = clientContext->getMaxNumThreadForExec();
+        auto degrees = Degrees(numNodesMap, mm);
+        auto coreValues = CoreValues(numNodesMap, mm);
+        auto currentFrontier = getPathLengthsFrontier(context, PathLengths::UNVISITED);
+        auto nextFrontier = getPathLengthsFrontier(context, 0);
+        auto frontierPair = std::make_unique<KCoreFrontierPair>(currentFrontier, nextFrontier,
+            numThreads, &degrees, &coreValues);
+        // Initialize starting nodes (all nodes) in the next frontier.
+        // When beginNewIteration, next frontier will become current frontier
+        frontierPair->setActiveNodesForNextIter();
+        frontierPair->getNextSparseFrontier().disable();
+        // Compute Degree
+        auto degreeEdgeCompute = std::make_unique<DegreeEdgeCompute>(&degrees);
+        auto computeState = GDSComputeState(std::move(frontierPair), std::move(degreeEdgeCompute),
+            sharedState->getOutputNodeMaskMap());
+        GDSUtils::runFrontiersUntilConvergence(context, computeState, graph, ExtendDirection::BOTH,
+            1 /* maxIters */);
+        // Compute Core values
+        auto removeVertexEdgeCompute = std::make_unique<RemoveVertexEdgeCompute>(&degrees);
+        computeState.edgeCompute = std::move(removeVertexEdgeCompute);
+        auto coreValue = 0u;
+        auto numNodes = graph->getNumNodes(clientContext->getTransaction());
+        auto numNodesComputed = 0u;
+        while (numNodes != numNodesComputed) {
+            // Compute current core value
+            while (true) {
+                std::atomic<common::offset_t> numActiveNodes;
```

Comment:
Is it always safe to not explicitly initialize this to 0? I feel it to be more readable too by setting it to 0.

Reply:
Done